karton/system/system.py

```diff
@@ -107,32 +121,118 @@ def gc_collect_tasks(self) -> None:
                     task.headers.get("receiver", "<unknown>"),
                 )
             if will_delete:
-                self.backend.delete_task(task)
-                self.backend.increment_metrics(
-                    KartonMetrics.TASK_GARBAGE_COLLECTED,
-                    task.headers.get("receiver", "unknown"),
-                )
+                to_delete.append(task)
             else:
                 running_root_tasks.add(task.root_uid)
+
+        start = time.time()
+        if to_delete:
+            self.log.info("Tasks: Deleting %s tasks", len(to_delete))
+            to_increment = [task.headers.get("receiver", "unknown") for tas in to_delete]
```

Comment:
Uhhh nice typo. Fortunately this is only for metrics keys, so doesn't invalidate any of my measures.